Library/Homebrew/test/rubocops/lines/generate_completions_spec.rb

```diff
@@ -0,0 +1,78 @@
+# typed: false
+# frozen_string_literal: true
+
+require "rubocops/lines"
+
+describe RuboCop::Cop::FormulaAudit do
+  describe RuboCop::Cop::FormulaAudit::GenerateCompletionsDSL do
+    subject(:cop) { described_class.new }
+
+    it "reports an offense when writing to a shell completions file directly" do
+      expect_offense(<<~RUBY, "/homebrew-core/Formula/foo.rb")
+        class Foo < Formula
+          name "foo"
+
+          def install
+            (bash_completion/"foo").write Utils.safe_popen_read(bin/"foo", "completions", "bash")
```

Comment:
It would be nice to test some of the other features in this out. For example, `base_name`, different shell flags, etc.

Reply:
Sure, I will add these cases as soon as I find time the next few days!